Neos.Media/Classes/Domain/Service/AssetSourceService.php

```diff
@@ -0,0 +1,179 @@
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Log\PsrSystemLoggerInterface;
+use Neos\Flow\Persistence\PersistenceManagerInterface;
+use Neos\Flow\ResourceManagement\Exception;
+use Neos\Flow\ResourceManagement\ResourceManager;
+use Neos\Media\Domain\Model\Asset;
+use Neos\Media\Domain\Model\AssetSource\AssetProxy\HasRemoteOriginalInterface;
+use Neos\Media\Domain\Model\AssetSource\AssetProxy\SupportsIptcMetadataInterface;
+use Neos\Media\Domain\Model\AssetSource\AssetSourceAwareInterface;
+use Neos\Media\Domain\Model\AssetSource\AssetSourceInterface;
+use Neos\Media\Domain\Model\ImportedAsset;
+use Neos\Media\Domain\Model\ImportedAssetManager;
+use Neos\Media\Domain\Repository\AssetRepository;
+use Neos\Media\Domain\Repository\ImportedAssetRepository;
+use Neos\Media\Domain\Strategy\AssetModelMappingStrategyInterface;
+use Neos\Media\Exception\AssetSourceServiceException;
+
+/**
+ * A service for Asset Sources and Asset Proxies
+ *
+ * @Flow\Scope("singleton")
+ */
+final class AssetSourceService
+{
+    /**
+     * @Flow\InjectConfiguration(path="assetSources")
+     * @var array
+     */
+    protected $assetSourcesConfiguration;
+
+    /**
+     * @Flow\Inject
+     * @var ImportedAssetManager
+     */
+    protected $importedAssetManager;
+
+    /**
+     * @Flow\Inject
+     * @var ImportedAssetRepository
+     */
+    protected $importedAssetRepository;
+
+    /**
+     * @Flow\Inject
+     * @var AssetModelMappingStrategyInterface
+     */
+    protected $assetModelMappingStrategy;
+
+    /**
+     * @Flow\Inject
+     * @var ResourceManager
+     */
+    protected $resourceManager;
+
+    /**
+     * @Flow\Inject
+     * @var AssetRepository
+     */
+    protected $assetRepository;
+
+    /**
+     * @Flow\Inject
+     * @var PersistenceManagerInterface
+     */
+    protected $persistenceManager;
+
+    /**
+     * @Flow\Inject
+     * @var PsrSystemLoggerInterface
+     */
+    protected $systemLogger;
+
+    /**
+     * @var AssetSourceInterface[]
+     */
+    private $assetSources = [];
+
+    /**
+     * @return AssetSourceInterface[]
+     */
+    public function getAssetsSources(): array
+    {
+        $this->initialize();
+        return $this->assetSources;
+    }
+
+    /**
+     * Import a specified asset from the given Asset Source
+     *
+     * @param string $assetSourceIdentifier Identifier of the asset source to import from, e.g. "neos"
+     * @param string $assetIdentifier The asset-source specific identifier of the asset to import
+     * @return ImportedAsset
+     * @throws AssetSourceServiceException
+     * @throws \Exception
+     */
+    public function importAsset(string $assetSourceIdentifier, string $assetIdentifier)
```

Comment:
Return type and annotation should be added

Reply:
Return type, sure! But which annotation is missing?